### PR TITLE
[produce] Add waiting for new app to appear

### PR DESCRIPTION
**Pull request by KrauseFx**
Fixes https://github.com/fastlane/fastlane/issues/5651
Fixes https://github.com/fastlane/fastlane/issues/4797

### CODE REVIEW

File: produce/lib/produce/itunes_connect.rb [hunk 1]
```diff
@@ -22,14 +22,25 @@ def create_new_app
...
+                                                              version: Produce.config[:app_version] || "1.0",
+                                                              sku: Produce.config[:sku].to_s, # might be an int
+                                                              bundle_id: app_identifier,
+                                                              bundle_id_suffix: Produce.config[:bundle_identifier_suffix],
+                                                              company_name: Produce.config[:company_name])
+
+        UI.crash!("Something went wrong when creating the new app on iTC") if generated_app["adamId"].to_s.empty?
+
         application = fetch_application
+        while application.nil?
```
Comment (asfalcone) [inline produce/lib/produce/itunes_connect.rb]: Should we timeout after some amount of time?

Comment (KrauseFx) [inline produce/lib/produce/itunes_connect.rb]: Great idea 👍

**Comment by KrauseFx** [comment]
Great feedback @asfalcone, addressed with https://github.com/fastlane/fastlane/pull/5688/commits/e93966a1ecf30a0a0eecd9f2609a5e50e13e6b97

**Comment by asfalcone** [comment]
Looks good. 👍

**Pull request merged**